app/controllers/admin/product_import_controller.rb

```diff
@@ -79,11 +79,9 @@ def check_spreadsheet_has_data(importer)
     end
 
     def save_uploaded_file(upload)
-      filename = 'import' + Time.zone.now.strftime('%d-%m-%Y-%H-%M-%S')
-      extension = '.' + upload.original_filename.split('.').last
-      directory = 'tmp/product_import'
-      Dir.mkdir(directory) unless File.exist?(directory)
-      File.open(Rails.root.join(directory, filename + extension), 'wb') do |f|
+      extension = upload.original_filename.split('.').last
```

Comment:
Sorry to be late to the party but Ruby's [.extname](https://ruby-doc.org/core-2.1.5/File.html#method-c-extname) feels better than this.

Reply:
The uploaded file at this point is an `ActionDispatch::Http::UploadedFile` object that doesn't respond to `.extname`. It only has the `.original_filename` attribute.

Reply:
Interesting, it's actually a static method and slightly more clever than split. https://apidock.com/ruby/File/extname/class
```diff
-      extension = upload.original_filename.split('.').last
+      extension = File.extname(upload.original_filename)
```